Instore fixes & updates (#258)
mercadopago/devsite-docs: guides/instore-payments/qr-payments/qr-gas-station.en.md

---
sites_supported:
  - mla
  - mlb
  - mlv
  - mco
  - mlm
  - global

---

# QR Payments in Gas Stations

## Flow

![QR Petrolera Flow](/images/mobile/qr-gas-flow.en.svg)

> 1. The user scans the QR code from the Mercado Pago or Mercado Libre app. The QR contains your URL with the information of the position where the sale was made.

Reviewer: ```suggestion
1. The user scans the QR code which is related to your URL that has the information of the position where the sale was made.```

7. With the payment identifier, you can [search](https://www.mercadopago.com.ar/developers/en/reference/payments/_payments_search/get/) the payment and continue with your internal processes.

   > If the status is `approved`, the payment must be credited. On the other hand, if it is `rejected`, the app will retry the payment requesting another means of payment.

8. That is all! Inform your customer that the payment was processed correctly.

## Concepts

First you must familiarize yourself with the following concepts since you will use them during the integration.

| Attribute      | Description                                                  |
| -------------- | ------------------------------------------------------------ |
| `ACCESS_TOKEN` | It is the [access token](https://www.mercadopago.com/mlm/account/credentials) of the Mercado Pago account where the payments will be accredited. |
| `COLLECTOR_ID` | It is the user number of the Mercado Pago account, it is the last 9 digits of your `access_token`, after the middle script. |

> | EXTERNAL_ID    | It is the unique identifier of the supplier. It is an alphanumeric code defined by you, it can not contain spaces or special characters and the uppercase of the lowercase is not distinguished. |

Reviewer: Fede, lo dejaría porque lo pueden usar las estaciones de terceros para filtrar los pagos por surtidor/shop desde el Front (cuando tengamos ese feature). En el caso de las propias (YPF, Shell) pueden poner el APIES. Hoy en día usan el external reference para eso y no es correcto. Que les parece? Pondría: "Es el identificador único de punto de venta ya sea surtidor, tienda, lubricantes, etc."

| SPONSOR_ID     | `COLLECTOR_ID` of a developer's Mercado Pago account. You must create an account by brand (YPF, Shell, Axion, etc). |

> | APIES          | Unique identifier of the service station.                    |

Reviewer: Coincido.

  - `transaction_amount`: Decimal. Total amount.
  - `store_id`: String. Unique identifier of the business (identifier of the service station or APIES).
  - `products`: Array. List of products purchased with the following attributes
    - `code`: String. Product code.
    - `quantity`: Integer or decimal. 
    - `unit_price`: Decimal. 
    - `unit`: String. Metric unit.
  - `cashier_identification`: Object. Employee data.
    - `type`: String. Document type.
    - `number`: String. ID number.
  - `period`: String. Period number.
  - `shift`: String. Shift number. 
  - `affinity_plan`: String. 

### Payment Object

```json
{
    "id": 420101010101,
    "external_reference": "internal id",
    "status": "approved",
    "status_detail": "accredited",
    ...
}
```

**Definitions**

- `id`: Unique ID generated by Mercado Pago (you will need it to refund a payment).

- `external_reference`: Same alphanumeric string that you added when creating the order.

- `status`: Payment status.

  - `approved`

  - `rejected`

  - `refunded`

  - `charged_back`

- `status_detail`: Detailed information about the current status or the reason for rejection.

> Consult the [full documentation](https://www.mercadopago.com.mx/developers/en/reference/payments/resource/) about this object in our API Reference.

Author: Redirecciona al dominio correcto si estás en otro site

| ------------------------------------------------------------ | ------------------------------------------------------------ |
| The user scans a valid QR code before placing the order.     | The app does not display an order.                           |
| The user scans a QR code with invalid parameters, meaning that it refers to a nonexistent account. | The app reports that an error has occurred.                  |
| The user scans a valid code, once the order has been placed and the sales order has been created. | La app muestra la orden.                                     |
| The user generates an approved payment.                      | The POS system receives the information of an approved payment. |
| The user generates an rejected payment.                      | The POS system receives the information of an rejected payment. |
| A return of a payment is made from the POS.                  | The return is impacted in the buyer's account.               |

### Errors

[Here](https://www.mercadopago.com.mx/developers/en/guides/payments/api/handling-responses/) you can find our error dictionary.

## Reports

> Consult the [complete documentation](https://www.mercadopago.com.ar/ayuda/herramienta-conciliacion_2116) about the Mercado Pago reports.

Author: Solo existe en .ar :(
